### DISCUSSION TITLE
stack of heterogenous views

### DISCUSSION THREAD

**Discussion by JimitSoni18** [lapce/floem#613] [Q&A] [answered]
how do I render an arbitrary number of items on a horizontal stack/row, where all elements are not necessarily the same type. For example, I want to make a file explorer breadcrumb where buttons and text "/" are interleaved, where buttons have directory names and clicking them takes you to that directory

**Comment by JimitSoni18** [comment]
I found that View is implemented for Box<dyn View>, but is there a different / better approach?

**Reply by dzhou121** [reply, marked as answer]
Box is the way. A built in trait `into_any` is just for this purpose. https://github.com/lapce/floem/blob/4dbecc86b8f375face40eec7171cf3ca3aa5c66a/src/view.rs#L107
